bin/redirects/redirect.swift

```diff
@@ -0,0 +1,780 @@
+    (part: "LanguageGuide", filename: "StringsAndCharacters.html", docc: "stringsandcharacters"),
+    (part: "LanguageGuide", filename: "Subscripts.html", docc: "subscripts"),
+    (part: "LanguageGuide", filename: "TheBasics.html", docc: "thebasics"),
+    (part: "LanguageGuide", filename: "TypeCasting.html", docc: "typecasting"),
+
+    (part: "ReferenceManual", filename: "AboutTheLanguageReference.html", docc: "aboutthelanguagereference"),
+    (part: "ReferenceManual", filename: "Attributes.html", docc: "attributes"),
+    (part: "ReferenceManual", filename: "Declarations.html", docc: "declarations"),
+    (part: "ReferenceManual", filename: "Expressions.html", docc: "expressions"),
+    (part: "ReferenceManual", filename: "GenericParametersAndArguments.html", docc: "genericparametersandarguments"),
+    (part: "ReferenceManual", filename: "LexicalStructure.html", docc: "lexicalstructure"),
+    (part: "ReferenceManual", filename: "Patterns.html", docc: "patterns"),
+    (part: "ReferenceManual", filename: "ReferenceManualPart.html", docc: "referencemanual"),
+    (part: "ReferenceManual", filename: "Statements.html", docc: "statements"),
+    (part: "ReferenceManual", filename: "Types.html", docc: "types"),
+    (part: "ReferenceManual", filename: "zzSummaryOfTheGrammar.html", docc: "summaryofthegrammar"),
+
+    (part: "RevisionHistory", filename: "RevisionHistory.html", docc: "revisionhistory"),
+    (part: "RevisionHistory", filename: "RevisionHistoryPart.html", docc: "revisionhistory"),
+]
+
+
+// Values to interpolate into the JavaScript
+var headingRedirects = ""
+var chapterRedirects = ""
+
+for case let (stableID, _, docc) in headings {
+    if docc == "" { continue }
+    headingRedirects += 
+        """
+            "#\(stableID)": "\(docc)",\n
+        """
+}
+
+for case let (part, filename, docc) in chapters {
+    chapterRedirects +=
+        """
+            "\(part)/\(filename)": "\(docc)",\n
+        """
+}
+
+for case let (part, filename, docc) in chapters {
+    let html = 
+    """
+    <html>
+    <head>
+        <title>The Swift Programming Language: Redirect</title>
+        <meta http-equiv="refresh" content="10; url=https://docs.swift.org/swift-book/documentation/the-swift-programming-language/\(docc)">
+    </head>
+    <body>
+    <div>
+    This content has moved; redirecting to the
+    <a href="https://docs.swift.org/swift-book/documentation/the-swift-programming-language/" id="redirect">new location</a>.
+    <noscript>
+    <aside>
+    This page requires JavaScript.
+    Please turn on JavaScript and refresh the page.
+    </aside>
+    </noscript>
+    </div>
+    <script>
+    const headingRedirects = {
+    \(headingRedirects)
+    }
+
+    const chapterRedirects = {
+    \(chapterRedirects)
+    }
+
+    const path = window.location.pathname;
+    const pathComponents = path.split("/");
+    const chapter = pathComponents[pathComponents.length - 1];
+
+    const hash = window.location.hash;
```

Comment:
```suggestion
    let hash = window.location.hash;
```

This should be a `let` binding instead, since you might assign to it a few lines later.